packages/webdriverio/src/utils/interception/devtools.ts

```diff
@@ -60,29 +64,32 @@ export default class DevtoolsInterception extends Interception {
                 /**
                  * match mock url
                  */
-                if (!Interception.isMatchingRequest(mock.url, request.url)) {
+                if (!Interception.isMatchingRequest(mock.url, event.request.url)) {
                     continue
                 }
 
                 /**
                  * Add statusCode and responseHeaders to request to be used in expect-webdriverio
                  */
-                request.statusCode = responseStatusCode
-                request.responseHeaders = { ...responseHeaders }
+                event.request.statusCode = event.responseStatusCode
+                event.request.responseHeaders = { ...responseHeaders }
 
                 /**
                  * match filter options
                  */
                 if (
-                    filterMethod(request.method, mock.filterOptions.method) ||
-                    filterHeaders(request.headers, mock.filterOptions.requestHeaders) ||
+                    filterMethod(event.request.method, mock.filterOptions.method) ||
+                    filterHeaders(event.request.headers, mock.filterOptions.requestHeaders) ||
                     filterHeaders(responseHeaders, mock.filterOptions.headers) ||
-                    filterRequest(request.postData, mock.filterOptions.postData) ||
-                    filterStatusCode(responseStatusCode, mock.filterOptions.statusCode)
+                    filterRequest(event.request.postData, mock.filterOptions.postData) ||
+                    filterStatusCode(event.responseStatusCode, mock.filterOptions.statusCode)
                 ) {
                     continue
                 }
 
+                mock.emit('request', event)
+
+                const { requestId, request, responseStatusCode } = event
```

Comment:
after emitting `request`, some parts of `event` could be changed, so we defining `requestId, request, responseStatusCode` here so we don't have to redefine those after `request` event emitted